### ISSUE TITLE
fatal: [localhost]: FAILED! when running the playbook on alpine

### ISSUE THREAD

**Issue by lwille** [ansible-ThoTeam/nexus3-oss#323]
`ls` on busybox doesn't support `--ignore` and fails when called from `nexus_install.yml`.

A platform-independent way would be to use the [`find` built-in module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/find_module.html#ansible-collections-ansible-builtin-find-module). I'm in the process of preparing and testing a pull request.

I realize that this project officially only supports CentOS and Ubuntu/Debian; however, Ansible builtins should be used instead of bare commands whenever possible. OS abstraction is why we're using Ansible in the first place.

log output:
```
TASK [/infra-nexus3-oss : Check if data directory is empty (first-time install)] *************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": ["ls", "--ignore=lost+found", "/nexus-data/"], "delta": "0:00:00.006817", "end": "2021-05-19 08:29:45.519334", "msg": "non-zero return code", "rc": 1, "start": "2021-05-19 08:29:45.512517", "stderr": "ls: unrecognized option: ignore=lost+found\nBusyBox v1.32.1 () multi-call binary.\n\nUsage: ls [-1AaCxdLHRFplinshrSXvctu] [-w WIDTH] [FILE]...\n\nList directory contents\n\n\t-1\tOne column output\n\t-a\tInclude entries which start with .\n\t-A\tLike -a, but exclude . and ..\n\t-x\tList by lines\n\t-d\tList directory entries instead of contents\n\t-L\tFollow symlinks\n\t-H\tFollow symlinks on command line\n\t-R\tRecurse\n\t-p\tAppend / to dir entries\n\t-F\tAppend indicator (one of */=@|) to entries\n\t-l\tLong listing format\n\t-i\tList inode numbers\n\t-n\tList numeric UIDs and GIDs instead of names\n\t-s\tList allocated blocks\n\t-lc\tList ctime\n\t-lu\tList atime\n\t--full-time\tList full date and time\n\t-h\tHuman readable sizes (1K 243M 2G)\n\t--group-directories-first\n\t-S\tSort by size\n\t-X\tSort by extension\n\t-v\tSort by version\n\t-t\tSort by mtime\n\t-tc\tSort by ctime\n\t-tu\tSort by atime\n\t-r\tReverse sort order\n\t-w N\tFormat N columns wide\n\t--color[={always,never,auto}]\tControl coloring", "stderr_lines": ["ls: unrecognized option: ignore=lost+found", "BusyBox v1.32.1 () multi-call binary.", "", "Usage: ls [-1AaCxdLHRFplinshrSXvctu] [-w WIDTH] [FILE]...", "", "List directory contents", "", "\t-1\tOne column output", "\t-a\tInclude entries which start with .", "\t-A\tLike -a, but exclude . and ..", "\t-x\tList by lines", "\t-d\tList directory entries instead of contents", "\t-L\tFollow symlinks", "\t-H\tFollow symlinks on command line", "\t-R\tRecurse", "\t-p\tAppend / to dir entries", "\t-F\tAppend indicator (one of */=@|) to entries", "\t-l\tLong listing format", "\t-i\tList inode numbers", "\t-n\tList numeric UIDs and GIDs instead of names", "\t-s\tList allocated blocks", "\t-lc\tList ctime", "\t-lu\tList atime", "\t--full-time\tList full date and time", "\t-h\tHuman readable sizes (1K 243M 2G)", "\t--group-directories-first", "\t-S\tSort by size", "\t-X\tSort by extension", "\t-v\tSort by version", "\t-t\tSort by mtime", "\t-tc\tSort by ctime", "\t-tu\tSort by atime", "\t-r\tReverse sort order", "\t-w N\tFormat N columns wide", "\t--color[={always,never,auto}]\tControl coloring"], "stdout": "", "stdout_lines": []}
```

**Comment by zeitounator** [comment]
I didn't blame on that file because I am to afraid to find I'm the guilty guy. Will definitely accept once correctly tested (as you can see the ci/cd pipeline is not so healthy). Thanks.

**Comment by zeitounator** [comment]
Hi @lwille 

I'm getting back to maintenance after a long pause for 2 main reason: 1) I was quite busy on other while the role was running like a charm on my side (deploying several times a month) and I had no special need for enhancements 2) the CI was down making reviewing PRs quite a pain (this has been fixed). Sorry for not getting in touch sooner.

---

Just to let you know I re-implemented your closed PR based on the same idea in #358. It was missing some conditionnals in task/main and the find itself could be optimized more specifically when the dir was populated (check only at depth=1 and return any found entries wether it's a file or folder). Thanks a lot for your contribution and once again sorry for the tremendous delay.

**Comment by lwille** [comment]
Hey @zeitounator, thanks for letting me know and keep up the great work!
